### DISCUSSION TITLE
Do I need to use Babylon.js's "immersive-ar" for WebAR?

### DISCUSSION THREAD

**Discussion by elifares** [jeeliz/jeelizFaceFilter#212] [Q&A] [answered]
### Background
I was able to get jeelizFaceFilter working with Babylon.js on my local environment on my laptop (7-year old MacBook Air) running on Google Chrome browser (Version 88+). My webcam is accessed and my face was detected.

### The problem
When I go through the Babylon.js [WebXR and AR documentation](https://doc.babylonjs.com/divingDeeper/webXR/webXRARFeatures#simple-scene-in-immersive-ar), they use `sessionMode: "immersive-ar"` when I try to use that, it leads to an error message in my browser console `Session mode "immersive-ar" not supported in browser`

### My question
How come jeelizeFaceFilter was able to access my camera and also detect my face and apply an object over it without using "immersive-ar" like in the Babylon.js documentation? Do I need "immersive-ar" if I want to implement sunglasses over my eyes?

Thanks in advance

**Comment by xavierjs** [comment, marked as answer]
Babylon.js immersive-ar session is relying on WebXR.
AFAIK WebXR is only implemented on Android mobile devices (not on IOS and not on desktop).
WebXR direct access to camera aligned video stream for computer vision is not guaranteed yet by the standard.

Jeeliz FaceFilter relies on MediaStream API, a part of WebRTC (formerly getUserMedia API). It is the easiest way to get the camera video feed, and it works everywhere.

You don't need immersive-ar to use Jeeliz FaceFilter. The best way is to use 2 canvases aligned using CSS, 1 behind to initialize Jeeliz FaceFilter and display the video stream, 1 atop to display the babylon.js scene with a transparent background.

**Reply by elifares** [reply]
Makes sense, thanks for explaining that!

**Reply by elifares** [reply]
I have another question though, in [AR Coffee example](https://github.com/jeeliz/jeelizAR), does that use WebXR? It works on my browser.

**Reply by xavierjs** [reply]
This AR Coffee example does  not rely on WebXR (this one: https://github.com/jeeliz/jeelizAR/tree/master/demos/threejs/ARCoffee).

In JeelizAR repository there are also 2 similar demos relying on JeelizAR for first coffee cup detection then on a third party tech for SLAM to do the tracking and positioning of the teacup pouring coffee (one is using 8thWall and the other WebXR). But JeelizAR is not maintained anymore and may not work on some devices (I have a similar library called WebAR.rocks.object which is better but I don't release it publicly).
